service/src/main/java/com/codeforcommunity/requester/S3Requester.java

```diff
@@ -178,21 +181,76 @@ private static String validateBase64ImageAndUploadToS3(
    */
   public static String validateUploadImageToS3LucyEvents(String eventTitle, String base64Encoding)
       throws BadRequestImageException, S3FailedUploadException {
-    String fileName = getImageFileNameWithoutExtension(eventTitle);
+    String fileName = getFileNameWithoutExtension(eventTitle, "_thumbnail");
     return validateBase64ImageAndUploadToS3(fileName, externs.getDirPublic(), base64Encoding);
   }
 
   /**
-   * Removes special characters, replaces spaces, and appends "_thumbnail".
+   * Removes special characters, replaces spaces, and appends a suffix.
    *
    * @param eventTitle the title of the event.
+   * @param suffix the suffix to be appended
    * @return the String for the image file name (without the file extension).
    */
-  public static String getImageFileNameWithoutExtension(String eventTitle) {
+  public static String getFileNameWithoutExtension(String eventTitle, String suffix) {
     String title =
         eventTitle.replaceAll(
             "[!@#$%^&*()=+./\\\\|<>`~\\[\\]{}?]", ""); // Remove special characters
     return title.replace(" ", "_").toLowerCase()
-        + "_thumbnail"; // The desired name of the file in S3
+        + suffix; // The desired name of the file in S3
+  }
+
+  /**
+   * Validate the given string encoding of HTML and upload it to the user upload S3 bucket.
```

Comment:
💭 Do we want to look into making sure the HTML is valid beyond being decodeable, e.g. making sure tags are properly closed, all tags are recognized, etc.? Not sure how much effort that would take and if it's worth doing /cc @chromium-52

Reply:
I think some form of validation would be nice like making sure the HTML is valid

In cases where we can't trust users' inputs (which, to be fair, is always so that even in our case, this is somewhat but not necessarily an overkill), we can use libraries like [jsoup](https://jsoup.org/) or [OWASP Java HTML Sanitizer](https://github.com/OWASP/java-html-sanitizer) for input validation